### DISCUSSION TITLE
Use openAI with KeyBERT?

### DISCUSSION THREAD

**Discussion by akarshghale** [MaartenGr/BERTopic#1410] [Q&A] [answered]
Hi,

can someone please let me know whether it is possible to use openAI embeddings with keyBERT?

**Comment by MaartenGr** [comment]
You can use OpenAI embeddings with BERTopic as described [here](https://maartengr.github.io/BERTopic/getting_started/embeddings/embeddings.html#openai). This will also be used when you use `KeyBERTInspired`.

**Reply by akarshghale** [reply]
Hi, Thanks for the reply.

Is it possible to give it access to the embeddings as a np array?

**Reply by MaartenGr** [reply]
You can pass custom embeddings to BERTopic with `.fit_transform(docs, embeddings=embeddings)`. So that could also be OpenAI embeddings. 

The word embeddings for `KeyBERTInspired` will be calculated within the class instead.

**Reply by akarshghale** [reply, marked as answer]
```
import openai
from keybert import KeyBERT, BaseEmbedder

class OpenAIEmbedder(BaseEmbedder):
    def __init__(self, model="text-embedding-ada-002"):
        openai.api_key = "sk-xxxxxxxxxxxxxxxxxxxxxxxxxxxx"
        self.model = model

    def embed(self, docs):
        response = openai.Embedding.create(
            input=docs,
            model=self.model
        )
        embeddings = [data["embedding"] for data in response["data"]]
        return embeddings

openai_embedder = OpenAIEmbedder()
kw_model = KeyBERT(embedding_model=openai_embedder)

```

something like this right?

**Reply by MaartenGr** [reply]
That would indeed be the solution, yes. Just make sure to post this on the right repository next time since I figured you were talking about BERTopic instead. Also, do note that there will be many API calls made since embeddings are made for both the documents as well for each individual word.

**Reply by akarshghale** [reply]
Got it! Thank You!  Posted here because there was no 'discussions' in KeyBERT.
